### PR TITLE
Add python version, update Makefile

### PR DESCRIPTION
**Pull request by hunytalk**
Analytics Pipeline Pull Request
--- Python 3 code changes for DE Pipelines

Make sure that the following steps are done before merging:

  - [ ] If you have a migration please contact data engineering team before merging.
  - [ ] Before merging run full acceptance tests suite and provide URL for the acceptance tests run.
  - [ ] A member of data engineering team has approved the pull request.

### CODE REVIEW

File: edx/analytics/tasks/launchers/remote.py [hunk 1]
```diff
@@ -45,6 +45,7 @@ def main():
     parser.add_argument('--workflow-profiler', choices=['pyinstrument'], help='profiler to run on the launch-task process', default=None)
     parser.add_argument('--wheel-url', help='url of the wheelhouse', default=None)
     parser.add_argument('--virtualenv-extra-args', help='additional arguments passed to virtualenv command when creating the virtual environment', default=None)
+    parser.add_argument('--python-version', help='Python version that will be used to run pipelines task', default=None)
```
Comment (pwnage101) [inline edx/analytics/tasks/launchers/remote.py]: The help text will need an example.  e.g., `3` or `3.6` or `/usr/bin/python3.6`?

Comment (hunytalk) [reply edx/analytics/tasks/launchers/remote.py]: Ok will add that.

File: Makefile [hunk 2]
```diff
@@ -45,6 +45,9 @@ reset-virtualenv:
 	# without bash, environment variables are not available
 	bash -c 'virtualenv --clear ${ANALYTICS_PIPELINE_VENV}/analytics_pipeline'
 
+reset-virtualenv-py3:
+	bash -c 'virtualenv --clear --python=$$(which python3.5) ${ANALYTICS_PIPELINE_VENV}/analytics_pipeline'
```
Comment (doctoryes) [inline Makefile]: Why the `python3.5` here instead of `python3`?

Comment (hunytalk) [reply Makefile]: I'll revert to python3.

File: edx/analytics/tasks/launchers/remote.py [hunk 3]
```diff
@@ -48,6 +48,9 @@ def main():
     parser.add_argument('--skip-setup', action='store_true', help='assumes the environment has already been configured and you can simply run the task')
     parser.add_argument('--package', action='append', help='pip install these packages in the pipeline virtual environment')
     parser.add_argument('--extra-repo', action='append', help="""additional git repositories to checkout on the cluster during the deployment""")
+    parser.add_argument('--python-version',
+                        help='Python version that will be used to run pipelines task e.g., 3 or 3.6 or /usr/bin/python3.6',
```
Comment (pwnage101) [inline edx/analytics/tasks/launchers/remote.py]: I was actually asking a question, is it 3 or 3.6 or the full path?  Whichever is correct, say that in the help text.

Comment (doctoryes) [inline edx/analytics/tasks/launchers/remote.py]: Isn't the only valid selection here `/usr/bin/python3.6`?

Comment (hunytalk) [reply edx/analytics/tasks/launchers/remote.py]: Ticket mentions for version no only, but i think complete path like `/usr/bin/python3.6` makes more sense here.

**Review by doctoryes — APPROVED** [review]

**Review by bmedx — APPROVED** [review]

**Comment by brianhw** [comment]
I'm not sure I understand the intent of having py2 and py3 versions of makefile target.  Isn't it just a matter of setting up the virtualenv to either use python2 or python3 at the time the virtualenv is created (mkvirtualenv -p /path/to/python), and then all the other targets will do the right thing when they run python?

**Pull request merged**